lumicks/pylake/population/dwelltime.py

```diff
@@ -7,23 +7,32 @@
 from deprecated.sphinx import deprecated
 
 
-@dataclass
+@dataclass(frozen=True)
 class DwelltimeBootstrap:
     """Bootstrap distributions for a dwelltime model.
 
-    This class is stored in the `DwelltimeModel.bootstrap` attribute
-    and should not be constructed manually.
+    This class should be initialized using :meth:`lk.DwelltimeModel.calculate_bootstrap()
+    <lumicks.pylake.DwelltimeModel.calculate_bootstrap>` and should not be constructed manually.
 
     Attributes
     ----------
-    _samples : np.ndarray
-        array of optimized model parameters for each bootstrap sample pull; shape is
-        [number of parameters, number of samples]
+    model : :class:`~lumicks.pylake.DwelltimeModel`
+        Original model sampled for the bootstrap distribution
+    amplitude_distributions : np.ndarray
+        Array of sample optimized amplitude parameters; shape is [number of components, number of samples]
+    lifetime_distributions : np.ndarray
+        Array of sample optimized lifetime parameters; shape is [number of components, number of samples]
     """
 
-    _samples: np.ndarray = field(default_factory=lambda: np.array([]), repr=False)
+    model: "DwelltimeModel"
+    amplitude_distributions: np.ndarray = field(default_factory=lambda: np.array([]), repr=False)
+    lifetime_distributions: np.ndarray = field(default_factory=lambda: np.array([]), repr=False)
```

Comment:
I'd prefer if these didn't have defaults.

The way it is set up now means that they'll have an incorrect shape compared to real data when `n_components > 1` which leads to not being able to rely on it being the correct shape.

One could consider polling `n_components` from the model, but I don't really see a need for creating this class empty often (probably never once `_bootstrap` is deprecated) so maybe just omitting the default and handling it in the single use case for it is best.

Reply:
good point. Indeed, I think once we remove the `bootstrap` property we can just raise on empty construction. Also added a specific test for the empty instance, just as a double check, I think it's nice to have. We can consider removing this after the property is gone.

 I left the fixups as separate commits for now but I'll squash before merging